controllers/oscmachine_keypair_controller.go

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/services/storage"
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope"
+)
+
+// getKeyPairResourceId return the keypairName from the resourceMap base on resourceName (tag name + cluster uid)
+func getKeypair(resourceName string, machineScope *scope.MachineScope) (string, error) {
+	keypairRef := machineScope.GetKeypairRef()
+	if keypairName, ok := keypairRef.ResourceMap[resourceName]; ok {
+		return keypairName, nil
+	} else {
+		return "", fmt.Errorf("%s does not exist", resourceName)
+	}
+}
+
+// checkKeypairFormatParameters check Keypair parameters format
+func checkKeypairFormatParameters(machineScope *scope.MachineScope) (string, error) {
+	machineScope.Info("Check Keypair parameters")
+	keypairSpec := machineScope.GetKeypair()
+	//netSpec.SetDefaultValue()
```

Comment:
Can/ will use default values.

Reply:
look at azure [SSHPublicKey ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/1cc4c2d84feff6a7d2749125f87ade26f6a973e6/exp/api/v1beta1/azuremachinepool_default.go#L29) for  to influence default Values

Reply:
Ok